Cabal/Distribution/Simple/Test.hs

```diff
@@ -110,7 +110,7 @@ test args pkg_descr lbi flags = do
     allOk <- summarizePackage verbosity packageLog
     writeFile packageLogFile $ show packageLog
 
-    let isCoverageEnabled = fromFlag $ configCoverage $ LBI.configFlags lbi
+    let isCoverageEnabled = LBI.libCoverage lbi && LBI.exeCoverage lbi
```

Comment:
This... actually doesn't seem like an improvement. Since this logic shows up repeatedly maybe there should be a function?